pre_commit/languages/pygrep.py

```diff
@@ -45,6 +45,31 @@ def _process_filename_at_once(pattern: Pattern[bytes], filename: str) -> int:
     return retv
 
 
+def _process_filename_by_line_negated(
+        pattern: Pattern[bytes],
+        filename: str,
+) -> int:
+    with open(filename, 'rb') as f:
+        for line in f:
+            if pattern.search(line):
+                return 0
+    output.write_line(filename)
+    return 1
+
+
+def _process_filename_at_once_negated(
+        pattern: Pattern[bytes],
+        filename: str,
+) -> int:
+    with open(filename, 'rb') as f:
+        contents = f.read()
+        match = pattern.search(contents)
+    if match:
+        return 0
+    output.write_line(filename)
+    return 1
```

Comment:
please write `if: ... else: ...` -- it is much easier for humans to see at a glance that they are disparate branches

also if you're doing to dedent the `if match` you might as well dedent `match = ...` as well